### PR TITLE
Migrate pages group 1

### PR DESCRIPTION
**Pull request by ghost**
### Fixed Issues
$ https://github.com/Expensify/App/issues/36274
PROPOSAL: https://github.com/Expensify/App/issues/36274#issuecomment-1943709707

### Tests

- [X] Verify that no errors appear in the JS console
1. Re-login in with an account that has personal chat and workspace.
2. Open any personal chat and select `Private notes` option.
3. Verify that the page can display loading state normally before the first rendering is completed.
4. Navigate to Workspace > Members > `Invite new members` page.
5. Verify that the page can load and filter based on input normally.

### QA Steps

- [X] Verify that no errors appear in the JS console

1. Re-login in with an account that has personal chat and workspace.
2. Open any personal chat and select `Private notes` option.
3. Verify that the page can display loading state normally before the first rendering is completed.
4. Navigate to Workspace > Members > `Invite new members` page.
5. Verify that the page can load and filter based on input normally.

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

0714f222-508d-450e-8fa5-f82a35d7d447
94157e17-bfea-4a77-9151-dfab6a3f799d

</details>

<details>
<summary>Android: mWeb Chrome</summary>

79e7bf80-94ae-4423-8405-e7197f2cb75a
bc813835-c941-4970-9b4a-43bf6924ece9

</details>

<details>
<summary>iOS: Native</summary>

bc42d01a-f5fe-4d98-8169-1b09097c630e

7983947e-aa7e-4625-8fe3-4dff440d3308

</details>

<details>
<summary>iOS: mWeb Safari</summary>

c0d10b2a-c85f-46ac-9aa4-45f7432c9c22
7a1f496e-c963-4587-a428-e428329babe5

</details>
<details>
<summary>MacOS: Chrome / Safari</summary>

885106aa-5e3e-44ab-b3e7-430fd8891d4c
4161e513-f5c7-4613-a321-49af26593169

</details>

<details>
<summary>MacOS: Desktop</summary>

1e32af2b-bb25-4ea2-a260-6972dfca447d
f06a5aee-22ed-47db-a413-bb52c980c7ad

</details>

### CODE REVIEW

**Comment by godofoutcasts94** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by godofoutcasts94** [comment]
Hey! Looks like a [PR](https://github.com/Expensify/App/pull/34897), just got recently merged which migrated the components that I have been working on which is `WorkspaceProfilePage.tsx` and `WorkspaceProfileCurrencyPage.tsx`. 

cc - @ntdiary

File: src/pages/LoadingPage.tsx [hunk 1]
```diff
@@ -1,36 +1,31 @@
...
+    onBackButtonPress?: () => void;
 
-const defaultProps = {
-    onBackButtonPress: undefined,
+    /** Title of the Loading page */
+    title: string;
 };
 
-function LoadingPage(props) {
+function LoadingPage({onBackButtonPress = undefined, title}: LoadingPageProps) {
```
Comment (blazejkustra) [inline src/pages/LoadingPage.tsx:15]: No need for undefined default values

File: src/pages/workspace/WorkspaceProfileCurrencyPage.tsx
Comment (blazejkustra) [inline src/pages/workspace/WorkspaceProfileCurrencyPage.tsx:1]: Why is this file added in this PR?

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/pages/LoadingPage.tsx (hunk 1 above)
Comment (godofoutcasts94) [reply src/pages/LoadingPage.tsx:15]: okay sure, I will make that change.

File: src/pages/workspace/WorkspaceProfileCurrencyPage.tsx
Comment (godofoutcasts94) [reply src/pages/workspace/WorkspaceProfileCurrencyPage.tsx:1]: Okay, I was not sure whether to add it or remove it. So, I added it will remove it asap.

Comment (godofoutcasts94) [reply src/pages/workspace/WorkspaceProfileCurrencyPage.tsx:1]: I have removed it @blazejkustra

File: src/pages/LoadingPage.tsx (hunk 1 above)
Comment (godofoutcasts94) [reply src/pages/LoadingPage.tsx]: Resolved @blazejkustra

**Comment by godofoutcasts94** [comment]
The `ProcessMoneyRequestHoldPage.tsx` was also merged in this [PR](https://github.com/Expensify/App/pull/36841) 3 days ago that is why it has been removed as I resolved the merge conflict which also removed this  - `ProcessMoneyRequestHoldPage.tsx` file  as well

**Review by blazejkustra — APPROVED** [review]

**Comment by godofoutcasts94** [comment]
Hey @blazejkustra ! I hope everything is fine now ?

**Comment by blazejkustra** [comment]
@godofoutcasts94 Yes, looks good to me!

**Comment by godofoutcasts94** [comment]
@ntdiary if you are available. Please review this PR and let's close it as soon as possible.

**Comment by godofoutcasts94** [comment]
> @godofoutcasts94 , a minor change, can you please rename the file to `SearchInputManager.ts`, because it doesn't contain React syntax. Also, I'll be running tests concurrently. :)

sure, I will do it since I was a bit skeptical about it. So, I renamed it as `tsx`

**Comment by godofoutcasts94** [comment]
> ## Reviewer Checklist
> * [ ]  I have verified the author checklist is complete (all boxes are checked off).
> * [ ]  I verified the correct issue is linked in the `### Fixed Issues` section above
> * [ ]  I verified testing steps are clear and they cover the changes made in this PR
>   
>   * [ ]  I verified the steps for local testing are in the `Tests` section
>   * [ ]  I verified the steps for Staging and/or Production testing are in the `QA steps` section
>   * [ ]  I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
>   * [ ]  I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
> * [ ]  I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
> * [ ]  I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
> * [ ]  I verified tests pass on **all platforms** & I tested again on:
>   
>   * [ ]  Android: Native
>   * [ ]  Android: mWeb Chrome
>   * [ ]  iOS: Native
>   * [ ]  iOS: mWeb Safari
>   * [ ]  MacOS: Chrome / Safari
>   * [ ]  MacOS: Desktop
> * [ ]  If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
> * [ ]  I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
>   
>   * [ ]  I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
>   * [ ]  I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
>   * [ ]  I verified that comments were added to code that is not self explanatory
>   * [ ]  I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
>   * [ ]  I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
>   * [ ]  I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
>   * [ ]  I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
>   * [ ]  I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
>   * [ ]  I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
> * [ ]  If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
> * [ ]  I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
> * [ ]  I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
> * [ ]  I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
> * [ ]  I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
> * [ ]  If a new component is created I verified that:
>   
>   * [ ]  A similar component doesn't exist in the codebase
>   * [ ]  All props are defined accurately and each prop has a `/** comment above it */`
>   * [ ]  The file is named correctly
>   * [ ]  The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
>   * [ ]  The only data being stored in the state is data necessary for rendering and nothing else
>   * [ ]  For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
>   * [ ]  Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
>   * [ ]  All JSX used for rendering exists in the render method
>   * [ ]  The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
> * [ ]  If any new file was added I verified that:
>   
>   * [ ]  The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
> * [ ]  If a new CSS style is added I verified that:
>   
>   * [ ]  A similar style doesn't already exist
>   * [ ]  The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
> * [ ]  If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
> * [ ]  If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
> * [ ]  If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
> * [ ]  If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
> * [ ]  If the PR modifies the form input styles:
>   
>   * [ ]  I verified that all the inputs inside a form are aligned with each other.
>   * [ ]  I added `Design` label so the design team can review the changes.
> * [ ]  If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
> * [ ]  If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
> * [ ]  I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.
> 
> ### Screenshots/Videos
> Android: Native
> Android: mWeb Chrome
> iOS: Native
> iOS: mWeb Safari
> MacOS: Chrome / Safari
> MacOS: Desktop

I will update the new checklist since the files have been changed, I will change it accordingly

**Comment by godofoutcasts94** [comment]
> I've already filled out my reviewer checklist. @godofoutcasts94, have you renamed the file? Also, considering the actual changes to the files, we need to ensure that the normal page becomes `PrivateNotesEditPage` and `WorkspaceInvitePage`, you can update the test steps :)

Yes yes that's what I was talking about regarding updating the tests steps.

**Comment by godofoutcasts94** [comment]
> I've already filled out my reviewer checklist. @godofoutcasts94, have you renamed the file? Also, considering the actual changes to the files, we need to ensure that the normal page becomes `PrivateNotesEditPage` and `WorkspaceInvitePage`, you can update the test steps :)

Renamed the file from `tsx` to `ts`

**Comment by godofoutcasts94** [comment]
> I've already filled out my reviewer checklist. @godofoutcasts94, have you renamed the file? Also, considering the actual changes to the files, we need to ensure that the normal page becomes `PrivateNotesEditPage` and `WorkspaceInvitePage`, you can update the test steps :)

Updated the test steps with new screenshots/videos. Now please review and let me know if everything is fine according to you @ntdiary.

**Comment by godofoutcasts94** [comment]
> @godofoutcasts94, here are the steps I suggested. You can just update with them, without re-uploading the video. :)
> 
> 1. Re-login in with an account that has personal chat and workspace.
> 2. Open any personal chat and select `Private notes` option.
> 3. Verify that the page can display loading state normally before the first rendering is completed.
> 4. Navigate to Workspace > Members > `Invite new members` page.
> 5. Verify that the page can load and filter based on input normally.

Done @ntdiary

**Review by ntdiary — APPROVED** [review]
LGTM. :shipit:

**Comment by godofoutcasts94** [comment]
> > > @godofoutcasts94, here are the steps I suggested. You can just update with them, without re-uploading the video. :)
> > > 
> > > 1. Re-login in with an account that has personal chat and workspace.
> > > 2. Open any personal chat and select `Private notes` option.
> > > 3. Verify that the page can display loading state normally before the first rendering is completed.
> > > 4. Navigate to Workspace > Members > `Invite new members` page.
> > > 5. Verify that the page can load and filter based on input normally.
> > 
> > 
> > Done @ntdiary. Hope I am not ruining things for you or maybe irritating you which may lead to issues for me.
> 
> Ah? Don't worry, collaboration is enjoyable, please feel free to submit more proposals and resolve more issues. 😄

Thanks a lot for such support.

File: src/pages/LoadingPage.tsx [hunk 2]
```diff
@@ -1,36 +1,31 @@
...
+type LoadingPageProps = {
     /** Method to trigger when pressing back button of the header */
-    onBackButtonPress: PropTypes.func,
-    title: PropTypes.string.isRequired,
-};
+    onBackButtonPress?: () => void;
 
-const defaultProps = {
-    onBackButtonPress: undefined,
+    /** Title of the Loading page */
```
Comment (blimpich) [inline src/pages/LoadingPage.tsx:11]: This comment feels unnecessary to me, can we get rid of it?

**Review by blimpich — CHANGES_REQUESTED** [review]
Looks good, one comment then I think we can merge.

File: src/pages/LoadingPage.tsx (hunk 2 above)
Comment (godofoutcasts94) [reply src/pages/LoadingPage.tsx:11]: Fine, I will fix it right away

**Review by blimpich — APPROVED** [review]
LGTM! Nice job!

**Comment by godofoutcasts94** [comment]
> LGTM! Nice job!

Thanks ! @blimpich
